fastlane_core/lib/fastlane_core/build_watcher.rb

```diff
@@ -42,24 +42,30 @@ def watching_build(app_id: nil, platform: nil)
       end
 
       def matching_build(watched_train_version: nil, watched_build_version: nil, app_id: nil, platform: nil)
-        matched_builds = Spaceship::TestFlight::Build.builds_for_train(app_id: app_id, platform: platform, train_version: watched_train_version, retry_count: 2)
-        matched_builds.find { |build| build.build_version == watched_build_version }
+        client = Spaceship::ConnectAPI::Base.client
+        build_deliveries = client.get_build_deliveries(filter: { app: app_id, cfBundleVersion: watched_build_version}, limit: 1)
+
+        # Actually only show processed builds
+        matched_builds = Spaceship::TestFlight::Build.all(app_id: app_id, platform: platform)
+        matched_build = matched_builds.find { |build| build.build_version.to_s == watched_build_version.to_s }
```

Comment:
Can this be done by a single API request?

Reply:
Actually.... I can hit `get_build_deliveries` and if it returns something than no need to call `Spaceship::TestFlight::Build.all` 🤔